gp-includes/things/translation.php

```diff
@@ -486,7 +486,10 @@ public function reject() {
 	}
 
 	public function set_status( $status ) {
-		if ( 'current' == $status ) {
+		if ( ( 'current' === $status || 'rejected' === $status ) && ! GP::$permission->current_user_can( 'approve', 'translation', $this->id, array( 'translation' => $this ) ) ) {
```

Comment:
Translators can reject their own translations, this change is going to break this.